contracts/libraries/StringUtils.sol

```diff
@@ -0,0 +1,79 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.13;
+
+import {strings} from "./strings.sol";
+
+// Extracted from https://github.com/zkemail/email-wallet-sdk/blob/main/src/helpers/StringUtils.sol
+library StringUtils {
+    using strings for *;
+
+    /**
+     * @dev Converts a hexadecimal string to a bytes32 value.
+     * @param hexStr The hexadecimal string to convert.
+     * @return result A bytes32 value representing the converted hexadecimal string.
+     */
+    function hexToBytes32(string calldata hexStr) public pure returns (bytes32 result) {
+        require(hexStr.toSlice().startsWith("0x".toSlice()), "invalid hex prefix");
+        hexStr = hexStr[2:];
+        require(bytes(hexStr).length == 64, "invalid hex string length");
+        uint256[] memory ints = hex2Ints(hexStr);
+        uint256 sum = 0;
+        for (uint256 i = 0; i < 32; i++) {
+            sum = (256 * sum + ints[i]);
+        }
+        return bytes32(sum);
+    }
+
+    /**
+     * @dev Converts a hexadecimal string to a bytes.
+     * @param hexStr The hexadecimal string to convert. Must start with "0x" prefix.
+     * @return A bytes array representing the converted hexadecimal string.
+     */
+    function hexToBytes(string calldata hexStr) public pure returns (bytes memory) {
```

Comment:
@SoraSuegami two questions:

1. What do you think about trying to re-use the same logic for both `hexToBytes` and `hexToBytes32`? We could wrap `hexToBytes32` around `hexToBytes`. A similar option would be to use similar logic in both functions, at the moment, the approach of both functions differs slightly. Imaging something like this for wrapping `hexToBytes`:
```solidity
    function hexToBytes32(string calldata hexStr) public pure returns (bytes32 result) {
        bytes memory result = hexToBytes(hexStr);
        require(result.length == 32, "invalid hex string length");
        return bytes32(result);
    }
```

2. If this is being audited as part of Clave's audit, should we integrate this utility in our repos? Or stick to using this repo if anyone else needs to pull this in?

Reply:
Also @aalimsahin feel free to comment on these questions too if you have an opinion

Reply:
I think this is a good idea. It would make more sense to reduce the conversion logic to 1.

Reply:
> What do you think about trying to re-use the same logic for both hexToBytes and hexToBytes32? We could wrap hexToBytes32 around hexToBytes. 

I agree with reusing the same logic for hexToBytes32, and moving these implementations to our library.